acceptance/api_v1_services_mutation_test.go

```diff
@@ -0,0 +1,787 @@
+				Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+				Expect(string(bodyBytes)).To(Equal("service 'bogus' not found\n"))
+			})
+
+			Context("Already bound", func() {
+				BeforeEach(func() {
+					bindAppService(app, service, org)
+				})
+
+				It("returns a 'conflict' when the service is already bound", func() {
+					response, err := Curl("POST",
+						fmt.Sprintf("%s/api/v1/orgs/%s/applications/%s/services",
+							serverURL, org, app),
+						strings.NewReader(fmt.Sprintf(`{ "name": "%s" }`, service)))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response).ToNot(BeNil())
+
+					defer response.Body.Close()
+					bodyBytes, err := ioutil.ReadAll(response.Body)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response.StatusCode).To(Equal(http.StatusConflict), string(bodyBytes))
+					Expect(string(bodyBytes)).To(Equal("service '" + service + "' already bound\n"))
+				})
+			})
+
+			It("binds the service", func() {
+				response, err := Curl("POST",
+					fmt.Sprintf("%s/api/v1/orgs/%s/applications/%s/services",
+						serverURL, org, app),
+					strings.NewReader(fmt.Sprintf(`{ "name": "%s" }`, service)))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response).ToNot(BeNil())
+
+				defer response.Body.Close()
+				bodyBytes, err := ioutil.ReadAll(response.Body)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(response.StatusCode).To(Equal(http.StatusOK), string(bodyBytes))
+				Expect(string(bodyBytes)).To(Equal(""))
+			})
+		})
+	})
+
+	Describe("DELETE api/v1/orgs/:org/applications/:app/services/:service", func() {
+		var app string
+		var service string
+
+		BeforeEach(func() {
+			service = newServiceName()
+			app = newAppName()
+		})
+
+		It("returns a 'not found' when the org does not exist", func() {
+			response, err := Curl("DELETE",
+				fmt.Sprintf("%s/api/v1/orgs/idontexist/applications/%s/services/%s",
+					serverURL, app, service),
+				strings.NewReader(""))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("Organization 'idontexist' does not exist\n"))
+		})
+
+		It("returns a 'not found' when the application does not exist", func() {
+			response, err := Curl("DELETE",
+				fmt.Sprintf("%s/api/v1/orgs/%s/applications/bogus/services/%s",
+					serverURL, org, service),
+				strings.NewReader(""))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+			Expect(string(bodyBytes)).To(Equal("application 'bogus' not found\n"))
+		})
+
+		Context("Needs application", func() {
```

Comment:
maybe `With("existing application", func() {` reads better?

Reply:
same for the rest of the `Needs...` occurrences in this file

Reply:
Can you point me to the documentation for `With()` blocks ? I am unable to find any.
I see only `Describe()` and `Context()`. I guess I could use `with` in the block descriptions, where pertinent.

Reply:
Turns out `With` only existed in my mind ( :brain: ), sorry for the confusion. Maybe `Context("when an application exists")` or `When("application exists")`, I don't know. "Needs application" immediately brings the question "who needs it?", it took me a while to understand it's the test that needs it (is it?).

Reply:
Commit 0b5fb5a2e8dbb8af0982f37b5ef03cf786d6129b